### DISCUSSION TITLE
When using `code-tools`, how to hide the `code-fold` marker and summary in the rendered html?

### DISCUSSION THREAD

**Discussion by rraadd88** [quarto-dev/quarto-cli#11661] [Q&A]
### Description

Hi, I am rendering a notebook like this:
```
quarto render notebook.ipynb --to html -M code-fold:true -M code-tools:true
```
In the output html, I get code-fold marker and summary for each notebook cell as shown below, which I would like to avoid:

![image](https://github.com/user-attachments/assets/2d18f0fa-cba3-4620-9ada-6acefa6a34bf)

Avoiding such repetition (without merging the cells in the notebook) would be good, but ideally I would like to hide these elements altogether because I can use `code-tools`.

So the question is how to hide the `code-fold` marker and summary in rendered html when using `code-tools`?
Thank you.

**Comment by mcanouil** [comment]
If you don't want the code, why display it (`echo: true`) and fold it (`code-fold: true`)? Use `echo: false` (https://quarto.org/docs/computations/execution-options.html).

I don't understand what you are trying to do here.

**Reply by rraadd88** [reply]
Hi, thank you for your response. 

In other words, I am trying to do following:
1. by default, I would like to hide the code,
2. then using `code-tools:true` I would like to show/hide *all* the code.

So the same rendered html can be used for dual purpose: (1) to only show the text and images (plots) without any trace of code (or markers where the code is), and (2) to show the code alongside.

First, I tried 
```
quarto render notebook.ipynb --to html -M code-tools:true
```
which did not show the `code-tools` button.

Then I tried `echo:false`, which you also suggested
```
quarto render notebook.ipynb --to html -M code-tools:true -M echo:false
```
it also did not show the button and removed the code altogether. This is not what I want to do.

Using `code-fold`,
```
quarto render notebook.ipynb --to html -M code-tools:true -M code-fold:true
```
I can both hide the code (#1) and show it using `code-tools` (#2).

But now the problem is the code-fold markers and summaries (see the image), which clutter the html making it non-ideal to be used to show text and images only.  

Therefore, I would like to to hide the code-fold marker and summary in the rendered html, when using code-tools.

I hope this could be possible.

Versions:
quarto: 1.5.55

**Reply by mcanouil** [reply]
Could you share a **small self-contained "working" (reproducible)** example to work with, _i.e._, a complete Quarto document or a Git repository? The goal is to make it as easy as possible for us to recreate your problem so that we can fix it: please help us help you! Thanks.

---

You can share a **self-contained "working" (reproducible)** Quarto document using the following syntax, _i.e._, using more backticks than you have in your document (usually four ` ```` `).
See <https://quarto.org/bug-reports.html#small-is-beautiful-aim-for-a-single-document-with-10-lines>.

If you have multiple files (and if it is **absolutely required** to have multiple files), please share as a Git repository.

<table>
<tr><th>R</th><th>Python</th></tr>
<tr><td>

`````md
````qmd
---
title: "Reproducible Quarto Document"
format: html
engine: knitr
---

This is a reproducible Quarto document.

{{< lipsum 1 >}}

```{r}
x <- c(1, 2, 3, 4, 5)
y <- c(1, 4, 9, 16, 25)

plot(x, y)
```

![An image]({{< placeholder 600 400 >}}){#fig-placeholder}

{{< lipsum 1 >}}

The end after @fig-placeholder.
````
`````

</td><td>

`````md
````qmd
---
title: "Reproducible Quarto Document"
format: html
engine: jupyter
---

This is a reproducible Quarto document.

{{< lipsum 1 >}}

```{python}
import matplotlib.pyplot as plt

x = [1, 2, 3, 4, 5]
y = [1, 4, 9, 16, 25]

plt.plot(x, y)
plt.show()
```

![An image]({{< placeholder 600 400 >}}){#fig-placeholder}

{{< lipsum 1 >}}

The end after @fig-placeholder.
````
`````

</td></tr>
</table>

Additionally and if not already given, please share the output of `quarto check` within a code blocks (*i.e.*, using three backticks ` ```txt `), see <https://quarto.org/bug-reports.html#check>.

**Reply by rraadd88** [reply]
[test.txt](https://github.com/user-attachments/files/18099937/test.txt)
Content:
````qmd
---
title: "Reproducible Quarto Document"
format: html
engine: jupyter
code-fold: true
code-tools: true
---

In the non-ideal case, code chunks below (mimicking cells in a jupyter notebook) would make a clutter with repetitions of the code-fold markers and summaries:

```python
import logging
```

```python
import pathlib
```

```python
import functools
```

```python
import textwrap
```

In the ideal case, 
1. by default, all the code would be hidden without the `code-fold` clutter,
2. then using `code-tools:true` I would 
show/hide all the code.

`quarto check` output:
```txt
Quarto 1.5.55
[✓] Checking versions of quarto binary dependencies...
      Pandoc version 3.2.0: OK
      Dart Sass version 1.70.0: OK
      Deno version 1.41.0: OK
      Typst version 0.11.0: OK
[✓] Checking versions of quarto dependencies......OK
[✓] Checking Quarto installation......OK
      Version: 1.5.55
      Path: /Applications/quarto/bin

[✓] Checking tools....................OK
      TinyTeX: v2024.07.03
      Chromium: (not installed)

[✓] Checking LaTeX....................OK
      Using: TinyTex
      Path: /Users/user/Library/TinyTeX/bin/universal-darwin
      Version: 2024

[✓] Checking basic markdown render....OK

[✓] Checking Python 3 installation....OK
      Version: 3.12.4 (Conda)
      Path: /Users/user/pkg_mgr/bin/python
      Jupyter: 5.7.2
      Kernels: python3

[✓] Checking Jupyter engine render....OK

[✓] Checking R installation...........OK
      Version: 4.3.3
      Path: /Users/user/pkg_mgr/lib/R
      LibPaths:
        - /Users/user/pkg_mgr/lib/R/library
      knitr: 1.48
      rmarkdown: (None)

      The rmarkdown package is not available in this R installation.
      Install with install.packages("rmarkdown")
```

````

If the issue is specifically with Jupyter notebook, the minimal example would look like this:

![image](https://github.com/user-attachments/assets/ccd04852-6535-4b4e-a6f9-758bfa0e474e)

**Reply by mcanouil** [reply]
I cannot reproduce your claim.

`echo: false` is the way to hide the code. https://quarto.org/docs/computations/execution-options.html
`code-fold: true` is the way to show and fold in a `<details>` tag. https://quarto.org/docs/output-formats/html-code.html#folding-code
`code-tools: true` is the way to show a button to see the source of the document as text (Jupyter Notebook are not text). https://quarto.org/docs/output-formats/html-code.html#code-tools

You can't go both ways, hide and show the code. `code-fold: true` implies `echo: true`.
There is no point to fold something you don't even want.
(you can hack your way but I don't see any reasons to simply not use the options as they are intended to be used)

all this is clearly described in the documentation and behave as expected/intended.

<table>
<tr><th>Input</th><th>Output</th></tr>
<tr><td>

````qmd
---
title: "Reproducible Quarto Document"
format: html
engine: jupyter
code-tools: true
execute: 
  echo: false
---

{{< lipsum 1 >}}

```{python}
print("Hello, world!")
```

{{< lipsum 1 >}}

```{python}
print("Goodbye, world!")
```
````

</td><td>

<img width="725" alt="image" src="https://github.com/user-attachments/assets/f8a8bde8-d56e-4fe1-9e7c-d7dbbcad3d66" />

<img width="724" alt="image" src="https://github.com/user-attachments/assets/05fe4638-191a-4198-a307-9ffabba2ff20" />


</td></tr>
</table>

If you want to include the original Notebook, there are other ways such as https://quarto.org/docs/output-formats/html-multi-format.html.

**Reply by rraadd88** [reply]
Hi again, I see, as I suspected, the issue is probably specific to jupyter notebooks (at least with my setup). Thank you for checking this though. I guess perhaps an option to place the `code-fold` stuff on the right side (as in the case of rendering by [rstudio](https://stackoverflow.com/q/64076708)) would have been great.

Anyway, for whoever stumbles on this, here's a hack to accomplish what I was trying to do: replace `<\head>` in the html with `<style>summary { display: none; }<\style><\head>`.

**Reply by mcanouil** [reply]
I don't understand your "hack", especially the "replace" part of it.
```qmd
include-in-header:
  - text: |
      <style>
      summary {
        display: none;
      }
      </style>
```

Or use `css` or `theme` YAML keys/options with appropriate files.

PS: I still don't get why you don't want to use `echo: false`.

**Reply by mcanouil** [reply]
Ok, there is a weird unexpected interaction which changes depending on how the options are called on a Jupyter Notebook.

Assuming:

````qmd
---
title: "Reproducible Quarto Document"
engine: jupyter
---

```{python}
print("Hello, world!")
```

```python
print("Goodbye, world!")
```
````

Converting it as a Jupyter Notebook via:

```sh
quarto convert index.qmd 
```


<table>
<tr><th>QMD</th><th>IPYNB</th></tr>
<tr><td>
Correct:

```sh
quarto render index.qmd --to html -M code-tools:true
```

<img width="250" alt="image" src="https://github.com/user-attachments/assets/64da7e48-1003-4c32-a7de-582ac93da4b0" />
</td><td>
Incorrect:

```sh
quarto render index.ipynb --to html -M code-tools:true
```

<img width="250" alt="image" src="https://github.com/user-attachments/assets/110e199e-2653-439b-a421-c88f9a2e670d" />
</td></tr>
<tr><td>
Correct:

```sh
quarto render index.qmd --to html -M code-tools:true -M echo:false
```

<img width="250" alt="image" src="https://github.com/user-attachments/assets/f92a6fcc-33f2-441a-a9a5-c7165126b6ff" />
</td><td>
Incorrect:

```sh
quarto render index.ipynb --to html -M code-tools:true -M echo:false
```

<img width="250" alt="image" src="https://github.com/user-attachments/assets/6eae4c62-8ab0-44b9-8cd7-e66c1436a7c6" />
</td></tr>
<tr><td>
Correct:

```sh
quarto render index.qmd --to html -M code-tools:true -M code-fold:true
```

<img width="250" alt="image" src="https://github.com/user-attachments/assets/b27c8f1c-2f0f-4af9-a784-db941fc889f7" />
</td><td>
Correct:

```sh
quarto render index.ipynb --to html -M code-tools:true -M code-fold:true
```

<img width="250" alt="image" src="https://github.com/user-attachments/assets/ec31d52c-3344-480e-977b-d218db5ab77b" />
</td></tr>
<tr><td>
Correct:

```sh
quarto render index.qmd --to html -M code-tools:true -M code-fold:true -M echo:false
```

<img width="250" alt="image" src="https://github.com/user-attachments/assets/3b5665e4-8066-4501-adbf-cdf513e8e192" />
</td><td>
Incorrect:

```sh
quarto render index.ipynb --to html -M code-tools:true -M code-fold:true -M echo:false
```
 
<img width="250" alt="image" src="https://github.com/user-attachments/assets/c6cfe984-3f4f-44e3-9549-09a6e5efe814" />
</td></tr>
</table>

**Reply by mcanouil** [reply]
I've opened an issue:
- #11682

@rraadd88 Thanks for the report
